src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.OrderService;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(shoppingCart.getProducts());
```

Comment:
from checklist: In completeOrder() use List.copyOf(shoppingCart.getProducts()) to copy products from shoppingCart to order and don’t forget to clear ShoppingCart after you complete user’s order.

Reply:
Thanks. Fixed.